src/trans-netlist/trans_to_netlist.cpp

```diff
@@ -168,6 +170,7 @@ literalt convert_trans_to_netlistt::new_input()
     symbol.type=bool_typet();
     symbol.is_input=true;
     symbol.base_name="input";
+    symbol.mode = mode;
```

Comment:
Perhaps change the lines above to use the `symbolt symbol{id, bool_typet{}, mode};` constructor.